mitosheet/mitosheet/step_performers/column_steps/change_column_dtype.py

```diff
@@ -39,174 +39,194 @@ class ChangeColumnDtypeStepPerformer(StepPerformer):
-                        column_header,
-                        old_dtype,
-                        new_dtype
-                    )
-                elif is_timedelta_dtype(new_dtype):
-                    raise make_invalid_column_type_change_error(
-                        column_header,
-                        old_dtype,
-                        new_dtype
-                    )
-            if is_int_dtype(old_dtype):
-                if is_bool_dtype(new_dtype):
-                    new_column = column.fillna(False).astype('bool')
-                elif is_int_dtype(new_dtype):
-                    pass
-                elif is_float_dtype(new_dtype):
-                    new_column = column.astype('float')
-                elif is_string_dtype(new_dtype):
-                    new_column = column.astype('str')
-                elif is_datetime_dtype(new_dtype):
-                    new_column = pd.to_datetime(
-                        column, 
-                        unit='s',
-                        errors='coerce'
-                    )
-                elif is_timedelta_dtype(new_dtype):
-                    new_column = to_timedelta_series(column)
-            elif is_float_dtype(old_dtype):
-                if is_bool_dtype(new_dtype):
-                    new_column = column.fillna(False).astype('bool')
-                elif is_int_dtype(new_dtype):
-                    new_column = column.fillna(0).astype('int')
-                elif is_float_dtype(new_dtype):
-                    pass
-                elif is_string_dtype(new_dtype):
-                    new_column = column.astype('str')
-                elif is_datetime_dtype(new_dtype):
-                    new_column = pd.to_datetime(
-                        column, 
-                        unit='s',
-                        errors='coerce'
-                    )
-                elif is_timedelta_dtype(new_dtype):
-                    new_column = to_timedelta_series(column)
-            elif is_string_dtype(old_dtype):
-                if is_bool_dtype(new_dtype):
-                    new_column = to_boolean_series(new_column)
-                elif is_int_dtype(new_dtype):
-                    new_column = to_int_series(column)
-                elif is_float_dtype(new_dtype):
-                    new_column = to_float_series(column)
-                elif is_string_dtype(new_dtype):
-                    pass
-                elif is_datetime_dtype(new_dtype):
-                    # Guess the datetime format to the best of Pandas abilities
-                    datetime_format = get_datetime_format(column)
-                    # If it's None, then infer_datetime_format is enough to figure it out
-                    if datetime_format is not None:
+
+        # Store the changed columns till the end, so we can change them in a single
+        # operation, so if a later change fails, it doesn't cause any issues
+        new_column_map = dict()
```

Comment:
Instead of doing this, can't we just move the try up to catch the error and then handle itself?

Reply:
Yeah!